lib/services/apparmor.pm

```diff
@@ -39,7 +39,13 @@ sub start_service {
 sub check_service {
     # SLE12-SP2 doesn't support systemctl to enable apparmor
     if (is_sle('12-SP3+', get_var('HDDVERSION'))) {
-        systemctl 'is-enabled apparmor.service';
```

Comment:
It is better to find the root cause why the service is not enabled at this time, then you can wait for the process or sth while not to wait all the times.

Reply:
The code is not wait but just sync.
It is hard to find out the root cause. Some time it might be aimed to cache some data IMO.

Reply:
> The code is not wait but just sync.
> It is hard to find out the root cause. Some time it might be aimed to cache some data IMO.

Actually if we find out the root cause fortunately, it is not a bug.

Reply:
I prefer you wait 2s here if it is enough while not sync, I don't think it is the method to solve or speed the service to be enabled.

Reply:
> I prefer you wait 2s here if it is enough while not sync, I don't think it is the method to solve or speed the service to be enabled.

You are right waiting 2s can work too. But the original error msg show us "Failed to get unit file state" so the sync can fix this issue as weel IMO.
"Sync" and "sleep $N" are both OK. I will keep the "sync"